packages/gatsby/src/schema/__tests__/node-model.js

```diff
@@ -239,9 +239,9 @@ describe(`NodeModel`, () => {
         expect(result.length).toBe(3)
       })
 
-      it(`creates page dependencies`, () => {
+      it(`does not create page dependencies for all nodes`, () => {
         nodeModel.getAllNodes({}, { path: `/` })
-        expect(createPageDependency).toHaveBeenCalledTimes(9)
+        expect(createPageDependency).not.toHaveBeenCalled()
```

Comment:
Not sure about this case. Before this PR we were adding `node` dependency for every single node when running `getAllNodes()` without a type. This is insane: some sites have millions of nodes which would translate into millions of those dependencies (and action dispatches).

In this PR I just disabled it (so we don't track at all) but the other option is to track all possible connection types instead. In this case, we will still invalidate every single query on any node change. That's probably a better option.

But ideally, we should just deprecate this method and remove it eventually.

Reply:
Given https://github.com/gatsbyjs/gatsby/pull/24754 and your suggestion of using connection - would adding `Node` interface (once `@nodeInterface` part is worked out) be enough to track all connections (as we "unroll" abstract types to concrete ones after #24754)?

Reply:
(and also tracking individual nodes will not invalidate query when completely new node is added without any changes to previous nodes, which actually should happen IMO)

Reply:
Great idea about tracking `Node` interface 👍